File: money-service/src/main/java/com/hmpay/money/application/port/in/RechargeMoneyCommand.java
package com.hmpay.money.application.port.in;


import com.hmpay.common.SelfValidating;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Value;

import jakarta.validation.constraints.NotNull;

@Value
@Builder
@EqualsAndHashCode(callSuper = false)
public
class RechargeMoneyCommand extends SelfValidating<RechargeMoneyCommand> {

    @NotNull
    private final String name;

    @NotNull
    private final String email;

    @NotNull
    private final String address;

    @NotNull
    private final boolean isValid;

    public RechargeMoneyCommand(String name, String email, String address, boolean isValid) {
        this.name = name;
        this.email = email;
        this.address = address;
        this.isValid = isValid;
    }
}

File: money-local-service/src/main/kotlin/com/hmpay/moneylocal/application/port/in/RechargeMoneyCommand.kt
package com.hmpay.moneylocal.application.port.`in`

import com.hmpay.common.SelfValidating
import jakarta.validation.constraints.NotNull

data class RechargeMoneyCommand(
    @field:NotNull val name: String,
    @field:NotNull val email: String,
    @field:NotNull val address: String,
    @field:NotNull val isValid: Boolean,
) : SelfValidating<RechargeMoneyCommand>()
